pkg/kubelet/kubelet.go

```diff
@@ -1722,12 +1752,41 @@ func (kl *Kubelet) makePodDataDirs(pod *api.Pod) error {
 	return nil
 }
 
+// syncPod is the transaction script for the sync of a single pod.
+//
+// Arguments:
+//
+// pod - the pod to sync
+// mirrorPod - the mirror pod for the pod to sync, if it is a static pod
+// podStatus - the current status (TODO: always from the status manager?)
+// updateType - the type of update (CREATE,ADD,DELETE,RECONCILE)
+//
+// The workflow is:
+//  1.  If the pod is being created, record pod worker start latency
+//  2.  Call generateAPIPodStatus to prepare an api.PodStatus for the pod
+//  3.  If the pod is being seen as running for the first time, record pod
+//      start latency
+//  4.  Update the status of the pod in the status manager
+//  5.  Kill the pod if it should not be running
+//  6.  Create a mirror pod if the pod is a static pod, and does not
```

Comment:
@Random-Liu is working on moving mirror pods management into a separate component.